src/mtev_http.c

```diff
@@ -1804,6 +1811,7 @@ _http_construct_leader(mtev_http_session_ctx *ctx) {
         }
       }
     }
+    if (!cl_present && keys != (const char **)static_key_array) free(keys);
```

Comment:
I think it might be needed. This is all enclosed in the following while loop:

while(!cl_present)

There's no guarantee that we get to the end of the loop with cl_present set. If we get to the end when "!cl_present" is true, we need to free the malloced data. Unless you're seeing something I'm not, I think it's safer to just include this check.